### PR TITLE
docs: update Emoji ID

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
Move to stable
Update the EmojiID to encode the network address using the checksum
Do Grammarly check

### CODE REVIEW

File: src/RFC-0152_EmojiId.md
Comment (AaronFeickert) [inline src/RFC-0152_EmojiId.md:84]: Probably want a colon `:` here instead.

File: src/RFC-0152_EmojiId.md
Comment (AaronFeickert) [inline src/RFC-0152_EmojiId.md:102]: I wouldn't use the term "encoded" for this, as I don't think it really conveys the nature of the modification. Maybe something like "masked" instead?

File: src/RFC-0152_EmojiId.md
Comment (AaronFeickert) [inline src/RFC-0152_EmojiId.md:109]: I'd be more specific about this, to avoid any confusion:
"... to compute an 8-bit masked checksum `C'` from `B` (described below)"

Otherwise, the casual reader might assume this is just the standard DammSum checksum.

File: src/RFC-0152_EmojiId.md
Comment (AaronFeickert) [inline src/RFC-0152_EmojiId.md:116]: "For the masked checksum we use..."

File: src/RFC-0152_EmojiId.md
Comment (AaronFeickert) [inline src/RFC-0152_EmojiId.md:118]: I would use a term like "assert" instead of "verify", since verification seems to imply part of the decoding and validation process.

File: src/RFC-0152_EmojiId.md
Comment (AaronFeickert) [inline src/RFC-0152_EmojiId.md:106]: Should mention that the 8-bit network encoding is part of the algorithm input here, along with the public key. Otherwise it's not clear where it comes into play.

File: src/RFC-0152_EmojiId.md
Comment (AaronFeickert) [inline src/RFC-0152_EmojiId.md:123]: "... from an emoji ID and expected network `N` as follows:"

File: src/RFC-0152_EmojiId.md
Comment (AaronFeickert) [reply src/RFC-0152_EmojiId.md:109]: I would also avoid the use of multiple symbols for the variable name `eC`, but that's just a nit and a personal preference.

**Review by AaronFeickert — CHANGES_REQUESTED** [review]

**Comment by AaronFeickert** [comment]
Should have also mentioned in the review that this approach technically reduces the effectiveness of the emoji ID. If a different network identifier is used during decoding and a transmission error occurs that mangles the emoji ID in the right way, it's possible for the checksum to pass when it shouldn't. This should probably be noted somewhere.

**Comment by AaronFeickert** [comment]
It might also be useful to indicate in the narrative _why_ the network byte is used, and to highlight that addresses intended for different use cases should have different network bytes.

Further, if it's in scope, it might be helpful to specify which network bytes Tari will use, or at least which network differentiations will be needed (project, address type, network).

File: src/RFC-0152_EmojiId.md
Comment (stringhandler) [inline src/RFC-0152_EmojiId.md:115]:
```suggestion
however, be sent over the wire in raw form and its far more likely that they will just be copied thus it's important to verify the
```

**Review by stringhandler — APPROVED** [review]

**Comment by stringhandler** [comment]
Just waiting for approval from @AaronFeickert, then can merge

File: src/RFC-0152_EmojiId.md
Comment (CjS77) [inline src/RFC-0152_EmojiId.md:51]:
```suggestion
This document describes the specification for Emoji Ids. Emoji Ids are encoded node ids used for humans to verify peer node addresses easily and for machines to verify that the address is being used in the correct context.
```

File: src/RFC-0152_EmojiId.md
Comment (CjS77) [inline src/RFC-0152_EmojiId.md:66]: The network id is masking the checksum, so is not encoded.

File: src/RFC-0152_EmojiId.md
Comment (CjS77) [inline src/RFC-0152_EmojiId.md:84]: Change language to past continuous tense.

```suggestion
The emoji have been selected such that:
```

File: src/RFC-0152_EmojiId.md
Comment (CjS77) [inline src/RFC-0152_EmojiId.md:86]: May as add the table of emoji here and their encoding values.

File: src/RFC-0152_EmojiId.md
Comment (CjS77) [inline src/RFC-0152_EmojiId.md:116]: Specifically, if you want to get technical, there's a ~4/256 = 1/64 chance that a transcription mistake leads to the incorrect network/version combination matching the checksum (assuming 4 ver/net combos).

**Review by CjS77 — COMMENTED** [review]
A few suggestions

**Comment by AaronFeickert** [comment]
I still think the encoding and algorithm descriptions aren't strictly correct as written (but maybe this is being too picky).

In the encoding algorithm, the step
```
Use the DammSum algorithm with `k = 8` and `m = 32` to compute an 8-bit masked checksum `'C` from `B` and network `N`.
```
isn't quite right, because DammSum doesn't produce masked checksums. Further, the algorithm doesn't specify that `N` is an input to it.

In the decoding algorithm, it also isn't specified that `N` is a required input.

The missing step in the encoding algorithm is to get the unmasked checksum from DammSum and then apply the masking. I'm a fan of making algorithm descriptions as clear as possible; if I were a first-time reader, I'd probably need to check an implementation to confirm what's actually going on (which reduces the effectiveness of the RFC).

To be specific, I'd change the encoding algorithm steps to be something like this:
```
The emoji ID is produced from a node public key `B` (serialized as 32 bytes) and a network identifier `N` (serialized as 8 bits) as follows:
- Use the DammSum algorithm with `k = 8` and `m = 32` to compute an 8-bit checksum `C` using `B` as input.
- Compute the masked checksum `C' = C XOR N`.
- Encode `B` into an emoji string using the emoji map.
- Encode `C'` into an emoji character using the emoji map.
- Concatenate `B` and `C'` as the emoji ID.
```

And I'd change the "preamble" of the decoding algorithm to be something like this:
```
The node public key is obtained from an emoji ID and a network identifier `N` (serialized a 8 bits) as follows:
```

**Pull request merged**
